File: GTiHub/GTiHub/Models/Source.cs
namespace GTiHub.Models.EntityModel
{
    #region

    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;

    #endregion

    public class Source : BaseEntity
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int SourceId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public bool Active { get; set; }
        public DateTime Effective_Date { get; set; }
       
        public  ICollection<ProjectSource> ProjectSources { get; set; }
        public  ICollection<SourceField> SourceFields { get; set; }      
    }
}

File: GTiHub/GTiHub/ClientApp/app/components/source/source.ts
export interface ISource {
    name: string;
    description: string;
    effective_Date: string;
    active: boolean;
    sourceFields?: ISourceField[];
    sourceId?: number;
}

export interface ISourceField {
    name: string;
    datatype: string;
    active: boolean;
    seqNum: number;
}

export class Source implements ISource {
    constructor(
        public name: string,
        public description: string,
        public effective_Date: string,
        public active: boolean,
        public seqNum: number,
        public sourceFields?: SourceField[],
        public sourceId?: number
    ) { }
}

export class SourceField implements ISourceField {
    constructor(
        public name: string,
        public datatype: string,
        public active: boolean,
        public seqNum: number) {}
}